nemo/core/connectors/save_restore_connector.py

```diff
@@ -0,0 +1,325 @@
+            self._save_state_dict_to_disk(model.state_dict(), model_weights)
+            self._make_nemo_file_from_folder(filename=save_path, source_dir=tmpdir)
+
+    def _default_restore_from(
+        self,
+        restore_path: str,
+        override_config_path: Optional[Union[OmegaConf, str]] = None,
+        map_location: Optional[torch.device] = None,
+        strict: bool = True,
+        return_config: bool = False,
+    ):
+        """
+		Restores model instance (weights and configuration) into .nemo file
+		Args:
+		restore_path: path to .nemo file from which model should be instantiated
+		override_config_path: path to a yaml config that will override the internal
+			config file or an OmegaConf / DictConfig object representing the model config.
+		map_location: Optional torch.device() to map the instantiated model to a device.
+			By default (None), it will select a GPU if available, falling back to CPU otherwise.
+		strict: Passed to load_state_dict. By default True
+		return_config: If set to true, will return just the underlying config of the restored
+			model as an OmegaConf DictConfig object without instantiating the model.
+
+		Example:
+			```
+			model = nemo.collections.asr.models.EncDecCTCModel.restore_from('asr.nemo')
+			assert isinstance(model, nemo.collections.asr.models.EncDecCTCModel)
+			```
+
+		Returns:
+		An instance of type cls or its underlying config (if return_config is set).
+		"""
+        app_state = AppState()
+
+        # Get path where the command is executed - the artifacts will be "retrieved" there
+        # (original .nemo behavior)
+        cwd = os.getcwd()
+
+        if map_location is None:
+            if torch.cuda.is_available():
+                map_location = torch.device('cuda')
+            else:
+                map_location = torch.device('cpu')
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            try:
+                self._unpack_nemo_file(path2file=restore_path, out_folder=tmpdir)
+                os.chdir(tmpdir)
+                if override_config_path is None:
+                    config_yaml = path.join(tmpdir, app_state.model_config_yaml)
+                else:
+                    # can be str path or OmegaConf / DictConfig object
+                    config_yaml = override_config_path
+                if not isinstance(config_yaml, (OmegaConf, DictConfig)):
+                    conf = OmegaConf.load(config_yaml)
+                else:
+                    conf = config_yaml
+                    if override_config_path is not None:
+                        # Resolve the override config
+                        conf = OmegaConf.to_container(conf, resolve=True)
+                        conf = OmegaConf.create(conf)
+                # If override is top level config, extract just `model` from it
+                if 'model' in conf:
+                    conf = conf.model
+
+                if return_config:
+                    instance = conf
```

Comment:
I still think that `restore_from` should always return instance of ModelPT derived object.

Hence there should be a separate method for "restoring configuration" only. Please refer to EFF `NeMoCookbook.retrieve_configuration`:
https://gitlab-master.nvidia.com/tlt/eff/-/blob/main/src/eff/cookbooks/nemo_cookbook.py#L450

Reply:
This is required by Nemo, EFF can chose to not expose this argument but it's mandatory requirement for Nemo

Reply:
Som, I understand that you need to retrieve config - this is ok. I am just saying that I suggest moving config retrieval to a separate method. This will increase the modularity of Connector.

Reply:
Not really. The connectors job is to save and restore. Restoring the config only is also it's job.